crates/task-impls/src/network.rs

```diff
@@ -211,6 +221,15 @@ impl<
                 Some(membership.get_leader(vote.current_view() + 1)),
             ),
 
+            // TODO GG `VidDiserpseSend` like `DAProposalSend` for now
+            SequencingHotShotEvent::VidDisperseSend(proposal, sender) => (
+                sender,
+                MessageKind::<TYPES, I>::from_consensus_message(SequencingMessage(Right(
+                    CommitteeConsensusMessage::VidDisperseMsg(proposal),
+                ))), // TODO GG not a CommitteeConsensusMessage
+                TransmitType::Broadcast, // TODO GG broadcast??
```

Comment:
This is an interesting case.  We'll want to only send 1 message to upload the shares to the CDN, but we'll need to "directly" message peers through libp2p to send each node their own unique share.  We should discuss more what the best way to do this is; I haven't given it much thought at the moment.  Perhaps the easiest solution is to handle the VID disperse message differently depending on which network implementation it is being routed through.  We can maybe use `MessagePurpose` for this.

Reply:
oh yes this is going to be painful. new issue https://github.com/EspressoSystems/HotShot/issues/1696